objay/src/main.c

```diff
@@ -0,0 +1,126 @@
+// md-toolchain example main.c
+// Michael Moffitt 2018
+//
+// This main shows a simple "hello world" demo.
+
+// megadrive.h is an umbrella for all headers in src/md. Specific modules like
+// md/vdp.h do not need to be individually included. However, utility funcitons
+// are not included, as they are not core support functions.
+#include "md/megadrive.h"
+
+#include "util/text.h"
+#include "util/plane.h"
+
+#define WALK_SPD 105
+#define WALK_ACCEL 4
+#define WALK_BRAKE 8
+#define LEFT_WALL 40
+#define RIGHT_WALL 280
+
+static unsigned int player_x;
+static signed short player_dx;
+static unsigned short player_frame;
+static unsigned short player_facing = 0;
+
+static void load_player(void) {
+//  LZ77UnCompVram(spritegfx_chrTiles, SPR_VRAM(16));
+  player_x = 64 << 8;
+  player_dx = player_frame = player_facing = 0;
+}
+
+static void move_player(void) {
+  unsigned short cur_keys = io_pad_read(0);
+  // Acceleration and braking while moving right
+  if (player_dx >= 0) {
+    if (cur_keys & BTN_RIGHT) {
+      player_dx += WALK_ACCEL;
+      if (player_dx > WALK_SPD) player_dx = WALK_SPD;
+      player_facing = 0;  // 0: R
+    } else {
+      player_dx -= WALK_BRAKE;
+      if (player_dx < 0) player_dx = 0;
+    }
+  }
+  
+  // Acceleration and braking while moving left
+  if (player_dx <= 0) {
```

Comment:
This is interesting - why did you not use an `else if` here? It seems to me that, if a player holds BTN_LEFT on the padm at the right time, they could coerce the player to do two frames' worth of movement if player_dx is >= 0 and < WALK_BRAKE and BTN_LEFT is held. The opposite is not true of BTN_RIGHT at the other crossover point.

This is not an important problem, but I thought it might be worth pointing out, because I may have missed something.

Reply:
I didn't use `else if` because if I did, Podge would never be able to accelerate to the left from a stop, as the branch for not already moving to the left would be executed instead.

You're right that if holding left during the last frame of braking, both the braking and the acceleration will be processed in the same frame. It could be fixed by moving acceleration processing to its own function that executes `return` on any braking. But it's a quirk that only a TASer would probably notice.